xray/test/test_backends.py

```diff
@@ -651,6 +658,24 @@ def roundtrip(self, data, **kwargs):
                 yield ds
 
 
+@requires_netCDF4
+class NetCDF4ClassicViaNetCDF4DataTest(CFEncodedDataTest, TestCase):
```

Comment:
I suppose it's a good idea to verify that everything can be roundtripped when using netCDF4-classic. But it doesn't directly test the change here, which is that float64 data can be roundtripped exactly with every netCDF format. Something like `test_roundtrip_float64_data` is probably a good way to solve that. You could put it next to these test method, then it will get tested with every backend: https://github.com/xray/xray/blob/6ed84a04334338533c6773ce4b37d2179130df18/xray/test/test_backends.py#L180-L197